internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -677,6 +698,34 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
+	if subnetID != nil {
+		// Wait for vnet to come back to Succeeded before releasing any locks
+		timeout, _ := ctx.Deadline()
+
+		stateConf := &pluginsdk.StateChangeConf{
```

Comment:
This logic exists in a few places, whilst this isn't a blocker for this instance - can we add a TODO to refactor this out into a separate `custompoller` so that we can reuse this in the future:

```suggestion
		// TODO: refactor this into a `custompoller` within the `network` package
		stateConf := &pluginsdk.StateChangeConf{
```